app/lib/frontend/handlers/admin.dart

```diff
@@ -8,13 +8,34 @@ import 'dart:async';
 
 import 'package:shelf/shelf.dart' as shelf;
 
+import '../../account/backend.dart';
 import '../../shared/handlers.dart';
 
 import '../backend.dart';
 import '../models.dart';
 import '../templates/admin.dart';
 import '../templates/misc.dart';
 
+/// Handles requests for /oauth/callback
+shelf.Response oauthCallbackHandler(shelf.Request request) {
+  final code = request.requestedUri.queryParameters['code'];
+  final state = request.requestedUri.queryParameters['state'];
+  if (code == null || state == null) {
+    return notFoundHandler(request);
+  }
+  final isWhitelisted = state.startsWith('/admin/confirm/');
+  if (isWhitelisted) {
+    return redirectResponse(request.requestedUri
+        .replace(
+          path: state,
+          queryParameters: request.requestedUri.queryParameters,
```

Comment:
I think we should do the invite acceptance step here... ie. add the user (now authenticated) to the list of uploaders for the package..

Then we should clear the cache for said package and redirect the user to the package page where the user will now appear listed under uploaders for the given package. Immediate feedback is good :)